unified-runtime/source/adapters/native_cpu/usm.cpp

```diff
@@ -15,12 +15,74 @@
 #include "context.hpp"
 #include <cstdlib>
 
+#ifdef __unix__
+#include <sys/mman.h>
+#endif
+
 namespace umf {
 ur_result_t getProviderNativeError(const char *, int32_t) {
   return UR_RESULT_ERROR_UNKNOWN;
 }
 } // namespace umf
 
+// Device flags are merged with host flags for native CPU since they are the
+// same target. If flags conflict, e.g. WRITE_ONLY | READ_ONLY, the behaviour is
+// unspecified.
+static inline ur_usm_host_mem_flags_t
+merge_flags(ur_usm_host_mem_flags_t HF, ur_usm_device_mem_flags_t DF) {
+  HF |= UR_USM_HOST_MEM_FLAG_INITIAL_PLACEMENT *
+        !!(UR_USM_DEVICE_MEM_FLAG_INITIAL_PLACEMENT & DF);
```

Comment:
This feels to me like it's trying to be too clever, I think this would be much more readable as
```c++
  if (DF & UR_USM_DEVICE_MEM_FLAG_INITIAL_PLACEMENT) {
    HF |= UR_USM_HOST_MEM_FLAG_INITIAL_PLACEMENT;
  }
```
and likewise for the rest.